src/scales/scale.time.js

```diff
@@ -36,10 +40,94 @@ module.exports = function(Chart) {
 			},
 		},
 		ticks: {
-			autoSkip: false
+			autoSkip: false,
+			mode: 'series',   // 'linear|series'
```

Comment:
`auto` is the actual `time` scale implementation. It currently generates linear ticks because that's the way how [`generateTicks`](https://github.com/chartjs/Chart.js/blob/master/src/helpers/helpers.time.js#L196) works but we can imagine options, for example to [exclude some ranges](2006) or whatever, that would make the generated ticks not linear anymore. In this case, `auto + series` would perfectly works and that's the reason I split it in 2 options since all combinations make sense.

By default having `auto + series` or `auto + linear` is the same as long as we don't change how `generateTicks` works. The question is, if the user switch to `ticks.source: 'labels'`, what result it should expect by default? I think `labels + series` would be the most wanted choice (which is your `timeseries` behavior).